### PR TITLE
feat: update&improve ssr in with-supertokens

### PR DESCRIPTION
**Pull request by porcellus**
## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: examples/with-supertokens/package.json [hunk 1]
```diff
@@ -7,9 +7,10 @@
   },
   "dependencies": {
     "next": "latest",
+    "prettier": "^2.7.1",
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
-    "supertokens-auth-react": "^0.21.2",
+    "supertokens-auth-react": "github:supertokens/supertokens-auth-react#next-breaking-change",
```
Comment (rishabhpoddar) [inline examples/with-supertokens/package.json]: need to change this to use npm. Keeping this comment open until we do

File: examples/with-supertokens/pages/auth/[[...path]].js [hunk 2]
```diff
@@ -18,15 +18,17 @@ export default function Auth() {
   }, [])
 
   return (
-    <div className={styles.container}>
-      <Head>
-        <title>SuperTokens 💫</title>
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+    <SuperTokensWrapper>
```
Comment (rishabhpoddar) [inline examples/with-supertokens/pages/auth/[[...path]].js]: Why do we need this here? We are surrounding the _app.js component with `SuperTokensWrapper` anyway?

File: examples/with-supertokens/pages/index.js [hunk 3]
```diff
@@ -37,9 +31,30 @@ export async function getServerSideProps(context) {
...
+    <div className={styles.container}>
+      <Head>
+        <title>SuperTokens 💫</title>
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <main className={styles.main}>
+        <h1 className={styles.title}>
+          Welcome to <a href="https://nextjs.org">Next.js!</a>
+        </h1>
+        <ThirdPartyEmailPasswordAuth requireAuth={true}>
```
Comment (rishabhpoddar) [inline examples/with-supertokens/pages/index.js]: We should keep this wrapped around the whole page component - it's easier to understand and see that this is being used.

File: examples/with-supertokens/pages/index.js [hunk 4]
```diff
@@ -37,9 +31,30 @@ export async function getServerSideProps(context) {
...
+          <ProtectedPage userId={props.userId} />
```
Comment (rishabhpoddar) [inline examples/with-supertokens/pages/index.js]: We should showcase the use of useSessionContext as well inside the protected page

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: examples/with-supertokens/pages/index.js (hunk 3 above)
Comment (porcellus) [reply examples/with-supertokens/pages/index.js]: I moved it out to showcase that SSR actually happens (and which parts get rendered)

File: examples/with-supertokens/pages/index.js [hunk 5]
```diff
@@ -37,13 +32,15 @@ export async function getServerSideProps(context) {
 
 export default function Home(props) {
   return (
-    <ThirdPartyEmailPasswordAuthNoSSR>
+    <ThirdPartyEmailPasswordAuth requireAuth={true}>
```
Comment (rishabhpoddar) [inline examples/with-supertokens/pages/index.js]: isn't requireAuth true the default here?

File: examples/with-supertokens/pages/index.js [hunk 6]
```diff
@@ -57,21 +54,31 @@ function ProtectedPage({ userId }) {
     }
   }
 
+  if (session.loading === true) {
+    // It should never come here, since this is wrapped by ThirdPartyEmailPasswordAuth with requireAuth=true
```
Comment (rishabhpoddar) [inline examples/with-supertokens/pages/index.js]: This comment is not needed.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**